traffic_ops/testing/api/v14/capabilities_test.go

```diff
@@ -0,0 +1,159 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import "testing"
+
+import "github.com/apache/trafficcontrol/lib/go-log"
+import "github.com/apache/trafficcontrol/lib/go-tc"
+
+// These capabilities are defined during the setup process in todb.go.
+// ANY TIME THOSE ARE CHANGED THIS MUST BE UPDATED.
+var staticCapabilities = []tc.Capability {
+	tc.Capability{
+		Name: "all-read",
+		Description: "Full read access",
+	},
+	tc.Capability {
+		Name: "all-write",
+		Description: "Full write access",
+	},
+	tc.Capability {
+		Name: "cdn-read",
+		Description: "View CDN configuration",
+	},
+}
+
+func TestCapabilities(t *testing.T) {
+	WithObjs(t, []TCObj{Capabilities}, func() {
+		ReplaceTestCapability(t)
+		GetTestCapabilities(t)
+	})
+}
+
+func CreateTestCapabilities(t *testing.T) {
+	for _,c := range testData.Capabilities {
+		resp, _, err := TOSession.CreateCapability(c)
+		log.Debugln("Response: ", c.Name, " ", resp)
+		if err != nil {
+			t.Errorf("could not create capability: %v", err)
+		}
+	}
+}
+
+func GetTestCapabilities(t *testing.T) {
+	testDataLen := len(testData.Capabilities) + len(staticCapabilities)
+	capMap := make(map[string]string, testDataLen)
+
+	for _,c := range testData.Capabilities {
+		capMap[c.Name] = c.Description
+		cap, _, err := TOSession.GetCapability(c.Name)
+		if err != nil {
+			t.Errorf("could not get capability '%s': %v", c.Name, err)
+			continue
+		}
+
+		if cap.Name != c.Name {
+			t.Errorf("requested capacity '%s' but got a capacity with the name '%s'", c.Name, cap.Name)
+		}
+		if cap.Description != c.Description {
+			t.Errorf("capacity '%s' has the wrong description, want '%s' but got '%s'", c.Name, c.Description, cap.Description)
+		}
+	}
+
+	// Hopefully this won't need to be done for much longer
+	for _,c := range staticCapabilities {
+		capMap[c.Name] = c.Description
+	}
+
+
+	caps, _, err := TOSession.GetCapabilities()
+	if err != nil {
+		t.Fatalf("could not get all capabilities: %v", err)
+	}
+	log.Debugln("capacities:", caps)
+	if len(caps) != testDataLen {
+		t.Errorf("response returned different number of capabilities than those that exist; got %d, want %d", len(caps), testDataLen)
+		return // we can't FATAL because this testing.T object is shared
+	}
+
+	for _,c := range caps {
+		if desc, ok := capMap[c.Name]; !ok {
+			t.Errorf("capability '%s' found in response, but not in test data!", c.Name)
+		} else {
+			if desc != c.Description {
+				t.Errorf("capability '%s' has description '%s' in response, but had '%s' in the test data", c.Name, c.Description, desc)
+			}
+			delete(capMap, c.Name)
+		}
+	}
+
+	for c,_ := range capMap {
+		t.Errorf("Capability '%s' existed in the test data but didn't appear in the response!", c)
+	}
+}
+
+func ReplaceTestCapability(t *testing.T) {
+	if len(testData.Capabilities) < 1 {
+		t.Skip("There aren't any test capabilities, skipping (this should probably be rectified!)")
```

Comment:
I wonder if we should fail the test is this case? if this happens something is causing it and we want to address it and I worry a skip message will just get lost in the big output